File: 0229-majority-element-ii/0229-majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        
        unordered_map<int,int>hash;
        int n = nums.size();
        for(auto a:nums){
            hash[a]++;
        }
        
        n = n/3;
        vector<int>ans;
        
        for(auto x:hash)
            if(x.second > n){
              ans.push_back(x.first);
            }
        
            return ans;
    
        
    }
};

File: 0229-majority-element-ii/0229-majority-element-ii.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        
      int n = nums.length;
       HashMap<Integer,Integer> mp = new HashMap<>();
      for(int x : nums)
      {
        mp.put(x , mp.getOrDefault(x,0)+1);
      }
      
      n = n/3;
      List<Integer> ans = new ArrayList<>();
      
      for(Map.Entry<Integer,Integer> entry : mp.entrySet())
      {
          if(entry.getValue() > n)
          {
              ans.add(entry.getKey());
          }
      }
      
      return ans;
    }
}